packages/flutter_tools/lib/src/persistent_tool_state.dart

```diff
@@ -63,6 +72,12 @@ class _DefaultPersistentToolState implements PersistentToolState {
 
   static const String _kFileName = '.flutter_tool_state';
   static const String _kRedisplayWelcomeMessage = 'redisplay-welcome-message';
+  static const List<String> _lastActiveVersion = <String>[
```

Comment:
What if this was a `Map<Channel, String>`, and then:

```
  String _versionKeyFor(Channel channel) => _lastActiveVersion[channel];
```

Reply:
Done